File: ui/src/components/stepforms/schemas/todurationnumber.ts
export default {
  $schema: 'http://json-schema.org/draft-07/schema#',
  title: 'Convert number column to duration step',
  type: 'object',
  properties: {
    name: {
      type: 'string',
      enum: ['todurationnumber'],
    },
    column: {
      type: 'string',
      minLength: 1,
      title: 'Column to convert',
      description: 'Column to convert',
      attrs: {
        placeholder: 'Select a column',
      },
    },
    unit: {
      type: 'string',
      enum: ['days', 'hours', 'minutes', 'seconds', 'milliseconds'],
    },
  },
  required: ['name', 'column', 'unit'],
  additionalProperties: false,
};

File: server/src/weaverbird/pipeline/steps/todurationnumber.py
from typing import Literal

from weaverbird.pipeline.steps.utils.base import BaseStep
from weaverbird.pipeline.types import ColumnName


class ToDurationNumberStep(BaseStep):
    name: Literal["todurationnumber"] = "todurationnumber"
    column: ColumnName
    unit: Literal["days", "hours", "minutes", "seconds", "milliseconds"]
